src/Optimizely/Notification/NotificationCenter.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Copyright 2017, Optimizely Inc and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Optimizely\Notification;
+
+use Monolog\Logger;
+use Exception;
+use Optimizely\Logger\LoggerInterface;
+use Optimizely\Logger\NoOpLogger;
+
+class NotificationCenter
+{
+    private $_notificationId;
+
+    private $_notifications;
+
+    private $logger;
+
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->_notificationId = 1;
+        $this->_notifications = [];
+        foreach(array_values(NotificationType::getAll()) as $type){
+            $this->_notifications[$type] = [];
+        }
+
+        $this->logger = $logger?: new NoOpLogger;
+    }
+
+    public function getNotificationId(){
+        return $this->_notificationId;
+    }
+
+    public function getNotifications(){
+        return $this->_notifications;
+    }
+
+    /**
+     * [addNotificationListener description]
+     * @param [type] $notification_type     [description]
+     * @param [type] $notification_callback [description]
+     */
+    public function addNotificationListener($notification_type, $notification_callback)
+    {
+        if (!NotificationType::isNotificationTypeValid($notification_type)) {
+            $this->logger->log(Logger::ERROR, "Invalid notification type.");
+            return null;
```

Comment:
what does it mean to return null? Can we throw exception or any error logger, this is how the user would know. Please check the trend if anything is mishandled how they throw exception or notify user.

Reply:
Python does not check for invalid type/callback. It simply adds to the list. 
I put it on my own. I think they our open to implementation variety where it sounds good. Our general behavior for invalid arguments is to log and return null. 
Please discuss this with Opz.

Reply:
ok, let me check, for now send this as it is.